models/plan_adquisicion.go

```diff
@@ -62,13 +65,47 @@ func ActualizarPlanAdquisicion(PlanAdquisicion map[string]interface{}, idStr str
 			"Publicado":     PlanAdquisicion["Publicado"],
 		}
 		error := request.SendJson(beego.AppConfig.String("plan_adquicisiones_crud_url")+"Plan_adquisiciones/"+idStr, "PUT", &PlanAdquisicionPut, PlanAdquisicionActualizar)
+		// logs.Debug("Plan PUT", PlanAdquisicionPut)
 		if error != nil {
 			return nil, error
 		} else {
+
+			idPlan := int(PlanAdquisicionAntiguo["Id"].(float64))
+
+			filtroJsonB := map[string]interface{}{
+				"Estado":              "Preliminar",
+				"PlanAdquisicionesId": idPlan,
+			}
+
+			data, _ := json.Marshal(filtroJsonB)
+
+			query := string(data)
+
+			// Se sugiere ordenar por fecha de modificación
+			sortby := "FechaModificacion"
+
+			// El orden descendente velará por traer el último registro modificado
+			order := "desc"
+
+			// Para traer el último
+			limit := "1"
+
+			if movimientoObtenido, err := movimientosCrud.GetMovimientoProcesoExterno(query, "", sortby, order, "", limit); err != nil {
+				logs.Error(err)
+				return nil, error
+			} else {
+				idMovimientoObtenido := int(movimientoObtenido.([]interface{})[0].(map[string]interface{})["Id"].(float64))
+				if _, err := movimientosCrud.CrearMovimientosPublicacion(idMovimientoObtenido); err != nil {
+					logs.Error(err)
+					return nil, error
+				}
+			}
```

Comment:
Si en el `if` siempre se retorna, se puede omitir el `else`
```suggestion
			}
			idMovimientoObtenido := int(movimientoObtenido.([]interface{})[0].(map[string]interface{})["Id"].(float64))
			if _, err := movimientosCrud.CrearMovimientosPublicacion(idMovimientoObtenido); err != nil {
				logs.Error(err)
				return nil, error
			}
```